File: Python/baseDatos/Serializador.py
import pickle
from gestorAplicacion.usuarios.cliente import Cliente
from gestorAplicacion.transacciones.bolsillo import Bolsillo
from gestorAplicacion.transacciones.pago import Pago
from gestorAplicacion.transacciones.transferencia import Transferencia


class Serializador:
    @classmethod
    def serializar(cls,tipo):
        if tipo=="Prestamos":
            pickfile = open("baseDatos/temp/prestamos.txt", "wb")
            serializa = Cliente.listaCuentas
            pickle.dump(serializa, pickfile,)
            pickfile.close

        elif tipo=="Bolsillos":
            pickfile = open("baseDatos/temp/bolsillos.txt", "wb")
            serializa = Cliente.listarCuentas().misBolsillos
            pickle.dump(serializa, pickfile)
            pickfile.close
        
        elif tipo=="Pagos":
            pickfile = open("baseDatos/temp/pagos.txt", "wb")
            serializa = Pago.getPagos()
            pickle.dump(serializa, pickfile)
            pickfile.close
        
        elif tipo=="Transferencias":
            pickfile = open("baseDatos/temp/transferencias.txt", "wb")
            serializa = Transferencia.lista
            pickle.dump(serializa, pickfile)
            pickfile.close
        
    def SerializarTodo():
        Serializador.serializar("Prestamos")
        Serializador.serializar("Bolsillos")
        Serializador.serializar("Pagos")
        Serializador.serializar("Transferencias")

File: Java/src/baseDatos/Serializador.java
package baseDatos;
import gestorAplicacion.transacciones.Bolsillo;
import gestorAplicacion.transacciones.Pago;
import gestorAplicacion.transacciones.Transferencia;
import gestorAplicacion.usuario.Cliente;

import java.io.*;

public class Serializador {
    private static File rutaTemp = new File("src\\baseDatos\\temp");

    public static void serializar(Cliente cliente) {
        FileOutputStream fos;
        ObjectOutputStream oos;
        File[] docs = rutaTemp.listFiles();
        PrintWriter pw;

        for (File file : docs) {
            try {
                pw = new PrintWriter(file);
            } catch (FileNotFoundException e) {
                e.printStackTrace();
            }
        }
        for (File file : docs) {
            if (file.getAbsolutePath().contains("prestamos.txt")) {
                try {
                    fos = new FileOutputStream(file);
                    oos = new ObjectOutputStream(fos);
                    oos.writeObject(Cliente.listaCuentas);
                } catch (FileNotFoundException e) {
                    e.printStackTrace();

                } catch (IOException e) {
                    e.printStackTrace();
                }
            } else if (file.getAbsolutePath().contains("bolsillos.txt")) {
                try {
                    fos = new FileOutputStream(file);
                    oos = new ObjectOutputStream(fos);
                    oos.writeObject(Cliente.listarCuentas().misBolsillos);
                } catch (FileNotFoundException e) {
                    e.printStackTrace();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }

            else if (file.getAbsolutePath().contains("pagos.txt")) {
                try {
                    fos = new FileOutputStream(file);
                    oos = new ObjectOutputStream(fos);
                    oos.writeObject(Pago.getPagos());
                } catch (FileNotFoundException e) {
                    e.printStackTrace();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
            else if (file.getAbsolutePath().contains("transferencias.txt")) {
                try {
                    fos = new FileOutputStream(file);
                    oos = new ObjectOutputStream(fos);
                    oos.writeObject(Transferencia.listatr);
                } catch (FileNotFoundException e) {
                    e.printStackTrace();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
    }
}